feature_tests/kotlin/somelib/src/main/kotlin/dev/diplomattest/somelib/CallbackWrapper.kt

```diff
@@ -0,0 +1,281 @@
+        }
+    }
+}
+internal interface Runner_DiplomatCallback_CallbackWrapper_test_multiple_cb_args_diplomatCallback_g: Callback {
+    fun invoke(ignored: Pointer?, arg0: Int ): Int
+}
+
+class DiplomatCallback_CallbackWrapper_test_multiple_cb_args_diplomatCallback_g_Native: Structure(), Structure.ByValue {
+    @JvmField
+    internal var dataa: Pointer = Pointer(0L);
+    @JvmField
+    internal var run_callback: Runner_DiplomatCallback_CallbackWrapper_test_multiple_cb_args_diplomatCallback_g
+        = object :  Runner_DiplomatCallback_CallbackWrapper_test_multiple_cb_args_diplomatCallback_g {
+                override fun invoke(ignored: Pointer?, arg0: Int ): Int {
+                    throw Exception("ERROR NOT IMPLEMENTED")
+                }
+            }
+    @JvmField
+    internal var destructor: Pointer = Pointer(0L);
+
+    // Define the fields of the struct
+    override fun getFieldOrder(): List<String> {
+        return listOf("dataa", "run_callback", "destructor")
+    }
+}
+
+class DiplomatCallback_CallbackWrapper_test_multiple_cb_args_diplomatCallback_g internal constructor (
+    internal val nativeStruct: DiplomatCallback_CallbackWrapper_test_multiple_cb_args_diplomatCallback_g_Native) {
+    val dataa: Pointer = nativeStruct.dataa
+    val run_callback: Callback = nativeStruct.run_callback
+    val destructor: Pointer = nativeStruct.destructor
+
+    companion object {
+        val NATIVESIZE: Long = Native.getNativeSize(DiplomatCallback_CallbackWrapper_test_multiple_cb_args_diplomatCallback_g_Native::class.java).toLong()
+        
+        fun fromCallback(cb: (Int)->Int): DiplomatCallback_CallbackWrapper_test_multiple_cb_args_diplomatCallback_g {
+            val callback: Runner_DiplomatCallback_CallbackWrapper_test_multiple_cb_args_diplomatCallback_g = object :  Runner_DiplomatCallback_CallbackWrapper_test_multiple_cb_args_diplomatCallback_g {
+                override fun invoke(ignored: Pointer?, arg0: Int ): Int {
+                    return cb(arg0);
+                }
+            }
+            val cb_wrap = DiplomatCallback_CallbackWrapper_test_multiple_cb_args_diplomatCallback_g_Native()
+            cb_wrap.run_callback = callback;
+            return DiplomatCallback_CallbackWrapper_test_multiple_cb_args_diplomatCallback_g(cb_wrap)
+        }
+    }
+}
+class CallbackWrapper internal constructor (
+    internal val nativeStruct: CallbackWrapperNative) {
+    val cantBeEmpty: Boolean = nativeStruct.cantBeEmpty > 0
+
+    companion object {
+        internal val libClass: Class<CallbackWrapperLib> = CallbackWrapperLib::class.java
+        internal val lib: CallbackWrapperLib = Native.load("somelib", libClass)
+        val NATIVESIZE: Long = Native.getNativeSize(CallbackWrapperNative::class.java).toLong()
+        
+        fun testMultiArgCallback(f: DiplomatCallback_CallbackWrapper_test_multi_arg_callback_diplomatCallback_f, x: Int): Int {
```

Comment:
issue: I think this should take `fn(Int) -> Int` and internally call fromCallback?

Reply:
We always want the user-facing APIs to have nice idiomatic types

also this means that the diplomatCallback can be private